### PR TITLE
Use direct download of proxy to fix Travis

### PR DESCRIPTION
**Pull request by schlessera**
This PR skips using `pip` (which is broken on old Travis images) to install the proxy and instead downloads it directly from https://mitmproxy.org/downloads/.

It also adapts the syntax of the `ps` command in the `start.sh` script for the proxy:

https://travis-ci.org/github/WordPress/Requests/jobs/767860940#L425

Finally, it adds a Travis-specific `start-travis.sh` file to launch the old binary needed on these images with the old arguments that don't work with recent versions of the proxy anymore.

Once we got rid of the Travis CI integration, the `tests/utils/proxy/start-travis.sh` file can be removed again.

### CODE REVIEW

**Comment by jrfnl** [comment]
Travis still appears to be broken, so it looks like this (alone) does not solve it.

File: .travis.yml [hunk 1]
```diff
@@ -5,11 +5,8 @@ jobs:
   fast_finish: true
   include:
    - php: 5.2
-     dist: precise
```
Comment (jrfnl) [inline .travis.yml:8]: These changes will definitely not work as Xenial doesn't support those PHP versions.

**Comment by schlessera** [comment]
Travis passes again with the above changes:

Build #1143 - WordPress_Requests - Travis CI - Google Chrome 2021-04-26 at 7 50 08 AM

https://travis-ci.org/github/WordPress/Requests/builds/768379650

**Comment by jrfnl** [comment]
Hero! 🎉 

Only one problem left: while the build was passing, the tests weren't actually being run. This was due to the `$PHPUNIT_BIN` not being defined since the GH Actions merge (mea culpa). I've added a commit which should hopefully fix this. 🤞

**Comment by jrfnl** [comment]
Tests are running and we have one failing test to fix.

**Comment by jrfnl** [comment]
Hmm.. actually two - one on PHP 5.2-5.4 and one for PHP 8.0 by the looks of it.

**Comment by jrfnl** [comment]
We have lift-off 🚀 

My last commit fixed the PHP 5.2-5.4 failure and the PHP 8.0 test failure has magically disappeared.

Shall we squash this in one atomic commit before merging ?

**Review by jrfnl — APPROVED** [review]

**Comment by schlessera** [comment]
Yay! We'll need to enable squash merges first, it seems...

**Pull request merged**